### PR TITLE
feat(advert-saga): [skp-124] created advert saga

### PR DESCRIPTION
**Pull request by doncevas**

### CODE REVIEW

File: apps/api/src/app/domains/adverts/commands/create-advert.command.ts [hunk 1]
```diff
@@ -4,22 +4,33 @@ import { ICommandHandler, EventPublisher, CommandHandler } from '@nestjs/cqrs';
...
 export class CreateAdvertHandler implements ICommandHandler<CreateAdvert> {
   @Inject() private readonly _publisher: EventPublisher;
+  @Inject() private readonly _usersService: UsersService;
 
-  async execute({ data }: CreateAdvert) {
+  async execute({ data, user }: CreateAdvert) {
+
+    const userFromDB = await this._usersService.getById(user._id.toHexString());
+    if (!userFromDB) {
+      throw new BadRequestException('This user dos`t exist');
```
Comment (Gardemarin) [inline apps/api/src/app/domains/adverts/commands/create-advert.command.ts:23]: Need write:   doesn`t

File: apps/api/src/app/models/users/users.module.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { UserSchema } from './user.schema';
 
 @Module({
   imports: [MongooseModule.forFeature([{ name: 'User', schema: UserSchema }])],
-  providers: [UsersService]
+  providers: [UsersService],
+  exports: [UsersService]
```
Comment (Gardemarin) [inline apps/api/src/app/models/users/users.module.ts:9]: Why UsersService used as providers and exports? (It's just question).

File: apps/api/src/app/models/users/users.service.ts [hunk 3]
```diff
@@ -72,11 +76,12 @@ export class UsersService {
   }
 
   async updateSettings(settingsDto: SettingsDto): Promise<User> {
-    const userFromDb = await this.userModel.findOne({ email: settingsDto.email});
+    const userFromDb = await this.model.findOne({ email: settingsDto.email});
     if(!userFromDb) throw new HttpException('COMMON.USER_NOT_FOUND', HttpStatus.NOT_FOUND);
     
     userFromDb.settings = userFromDb.settings || {};
     for (const key in settingsDto) {
+      // eslint-disable-next-line no-prototype-builtins
```
Comment (Gardemarin) [inline apps/api/src/app/models/users/users.service.ts:84]: What means this comment?

Comment (doncevas) [reply apps/api/src/app/models/users/users.service.ts:84]: > What means this comment?

don't use it :D on this case type script don't understand hasOwnProperty, and this line disabling eslint error for current line

**Pull request merged**